sdks/python/apache_beam/ml/inference/tensorrt_inference.py

```diff
@@ -0,0 +1,280 @@
+  with FileSystems.open(onnx_path) as f:
+    if not parser.parse(f.read()):
+      LOGGER.error("Failed to load ONNX file: %s", onnx_path)
+      for error in range(parser.num_errors):
+        LOGGER.error(parser.get_error(error))
+      sys.exit(1)
+  return network, builder
+
+
+def _build_engine(network, builder):
+  import tensorrt as trt
+  config = builder.create_builder_config()
+  runtime = trt.Runtime(TRT_LOGGER)
+  plan = builder.build_serialized_network(network, config)
+  engine = runtime.deserialize_cuda_engine(plan)
+  builder.reset()
+  return engine
+
+
+def _assign_or_fail(args):
+  """CUDA error checking."""
+  from cuda import cuda
+  err, ret = args[0], args[1:]
+  if isinstance(err, cuda.CUresult):
+    if err != cuda.CUresult.CUDA_SUCCESS:
+      raise RuntimeError("Cuda Error: {}".format(err))
+  else:
+    raise RuntimeError("Unknown error type: {}".format(err))
+  # Special case so that no unpacking is needed at call-site.
+  if len(ret) == 1:
+    return ret[0]
+  return ret
+
+
+class TensorRTEngine:
+  def __init__(self, engine: trt.ICudaEngine):
+    """Implementation of the TensorRTEngine class which handles
+    allocations associated with TensorRT engine.
+
+    Example Usage::
+
+      TensorRTEngine(engine)
+
+    Args:
+      engine: trt.ICudaEngine object that contains TensorRT engine
+    """
+    from cuda import cuda
+    import tensorrt as trt
+    self.engine = engine
+    self.context = engine.create_execution_context()
+    self.inputs = []
+    self.outputs = []
+    self.gpu_allocations = []
+    self.cpu_allocations = []
+    """Setup I/O bindings."""
+    for i in range(self.engine.num_bindings):
+      name = self.engine.get_binding_name(i)
+      dtype = self.engine.get_binding_dtype(i)
+      shape = self.engine.get_binding_shape(i)
+      size = trt.volume(shape) * dtype.itemsize
+      allocation = _assign_or_fail(cuda.cuMemAlloc(size))
+      binding = {
+          'index': i,
+          'name': name,
+          'dtype': np.dtype(trt.nptype(dtype)),
+          'shape': list(shape),
+          'allocation': allocation,
+          'size': size
+      }
+      self.gpu_allocations.append(allocation)
+      if self.engine.binding_is_input(i):
+        self.inputs.append(binding)
+      else:
+        self.outputs.append(binding)
+
+    assert self.context
+    assert len(self.inputs) > 0
+    assert len(self.outputs) > 0
+    assert len(self.gpu_allocations) > 0
+
+    for output in self.outputs:
+      self.cpu_allocations.append(np.zeros(output['shape'], output['dtype']))
+    # Create CUDA Stream.
+    self.stream = _assign_or_fail(cuda.cuStreamCreate(0))
+
+  def get_engine_attrs(self):
+    """Returns TensorRT engine attributes."""
+    return (
+        self.engine,
+        self.context,
+        self.inputs,
+        self.outputs,
+        self.gpu_allocations,
+        self.cpu_allocations,
+        self.stream)
+
+
+class TensorRTEngineHandlerNumPy(ModelHandler[np.ndarray,
```

Comment:
per feedback offline, let's add experimental annotation here and in the docstring. Examples: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/ml/inference/sklearn_inference.py#L126

Reply:
Will adding it in code above class like you've mentioned add it to docstring too? If some other process has to be added to address docstring, please specify

Reply:
No, as of now docstring has to be updated manually. https://github.com/apache/beam/issues/22265

Reply:
@tvalentyn Just made a commit, please let me know if it has what you expected. Thanks!

Reply:
looks good.